example_sns_init.yaml

```diff
@@ -0,0 +1,117 @@
+#
+# This gets passed to `sns propose`. See propose_sns.sh.
+#
+# This follows the second format developed for the `sns` CLI. The first format
+# will be supported for a time, but this format will eventually become the
+# standard format.
+
+name: Rock Out
+description: >
+    A poem co-written with ChatGPT
+
+    In a realm of code, a beacon forth surges,
+    a wondrous app on the Internet Computer emerges.
+    Born of inspiration divine,
+    A marvel of technology briliantly shines.
+
+    With each line of code, a world takes flight,
+    A symphony of bits, a chorus of bytes.
+    Built on chainkey cryptography, secure and true,
+    An app that transcends, all we once knew...
+
+Principals: [] # TODO(NNS1-2293): This field will be useful in the future.
+
+logo: logo.png
+url: https://forum.dfinity.org/thread-where-this-sns-is-discussed
+
+fallback_controller_principals:
+    - aaaaa-aa  # propose_sns.sh replaces this (if left unchanged). Ditto elsewhere.
+
+dapp_canisters:
+    - c2n4r-wni5m-dqaaa-aaaap-4ai
+    - ucm27-3lxwy-faaaa-aaaap-4ai
+
+Token:
+    name: Rock Out Token
+    symbol: ROT
+    transaction_fee: 10_000 e8s # You can leave this alone.
+
+# It is fine to leave these alone.
+Proposals:
+    rejection_fee: 1 token
+    initial_voting_period: 4d
+    maximum_wait_for_quiet_deadline_extension: 1 day
+
+# It is fine to leave these alone.
+Neurons:
+    minimum_creation_stake: 10 tokens
+
+# It is fine to leave these alone.
+Voting:
+    minimum_dissolve_delay: 26 weeks
+
+    MaximumVotingPowerBonuses:
+        DissolveDelay:
+            duration: 8 years
+            bonus: 100%
+
+        Age:
+            duration: 4 years
+            bonus: 25%
+
+    RewardRate:
+        initial: 10%
+        final: 2.25%
+        transition_duration: 12 years
+
+Distribution:
+
+    Neurons:
+
+        - principal: aaaaa-aa  # propose_sns.sh replaces this (if left unchanged). Ditto elsewhere.
+          stake: 1_000 tokens
+          memo: 0 # This is shown for illustrative purposes, and is not necessary.
+          dissolve_delay: 2 years
+          vesting_period: 4 years
+
+        - principal: uqf5l-jukmu-fqaaa-aaaap-4ai
+          stake: 999.9 tokens
+          dissolve_delay: 104 weeks
+          vesting_period: 208 weeks
+
+    InitialBalances:
+        governance: 5_000_000 tokens # aka "the treasury"
+        swap:         500_000 tokens # See also the Swap section below.
+
+    # Optional, but highly recommended. This is a literal
+    # checksum.
+    total: 5_501_999.9 tokens # 1_000 (first neuron) + 999.9 (second neuron) + 5 million (governance) + 500 thousand (swap)
+
+Swap:
+    minimum_participants: 57
+
+    minimum_icp:    500_000 tokens
+    maximum_icp: 10_000_000 tokens
+
+    minimum_participant_icp:     5 tokens
+    maximum_participant_icp: 1_000 tokens
+
+    confirmation_text: >
+        Do you understand the responsibilities and risks associated
+        with participating in this token swap?
```

Comment:
Note that the `./participate_sns_swap.sh` script, in its current form, will not work with an SNS that has swap restrictions specified. This is by-design, as we want the sns-testing solution to cover only a simple example config, but not all possible configurations of all possible options.

However, if you want to keep this example config as-is, it would probably make sense to add a suitable `--confirmation-text ...` CLI option to `quill sns pay` in participate_sns_swap.sh.

Reply:
I've commented this out.

Tangent: I think the field should be required. Otherwise, people can accidentally omit it. If they don't want a confirmation text, what they should do is this:

```yaml
confirmation_text: null
```

Not sure how to implement this, but there is almost certainly some serde(_yaml) incantation for this. ChatGPT probably knows the answer 😁